elfutils-0.188: fails to compile against latest curl (7.87.0)

### Steps To Reproduce
Steps to reproduce the behavior:
1. build elfutils-0.188

### Build log
```
debuginfod-client.c: In function 'debuginfod_query_server':
debuginfod-client.c:1330:7: error: 'CURLOPT_PROTOCOLS' is deprecated: since 7.85.0. Use CURLOPT_PROTOCOLS_STR [-Werror=deprecated-declarations]
 1330 |       curl_easy_setopt_ck(data[i].handle, CURLOPT_PROTOCOLS,
      |       ^~~~~~~~~~~~~~~~~~~
In file included from debuginfod-client.c:94:
/nix/store/arrxja9h6jxh9qs5iw3vxcznlkq0w00i-curl-7.87.0-dev/include/curl/curl.h:1749:3: note: declared here
 1749 |   CURLOPTDEPRECATED(CURLOPT_PROTOCOLS, CURLOPTTYPE_LONG, 181,
      |   ^~~~~~~~~~~~~~~~~
debuginfod-client.c:1468:11: error: 'CURLINFO_CONTENT_LENGTH_DOWNLOAD' is deprecated: since 7.55.0. Use CURLINFO_CONTENT_LENGTH_DOWNLOAD_T [-Werror=deprecated-declarations]
 1468 |           curl_res = curl_easy_getinfo(target_handle,
      |           ^~~~~~~~
In file included from debuginfod-client.c:94:
/nix/store/arrxja9h6jxh9qs5iw3vxcznlkq0w00i-curl-7.87.0-dev/include/curl/curl.h:2853:3: note: declared here
 2853 |   CURLINFO_CONTENT_LENGTH_DOWNLOAD
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
debuginfod-client.c:1503:15: error: 'CURLINFO_SIZE_DOWNLOAD' is deprecated: since 7.55.0. Use CURLINFO_SIZE_DOWNLOAD_T [-Werror=deprecated-declarations]
 1503 |               curl_res = curl_easy_getinfo(target_handle,
      |               ^~~~~~~~
In file included from debuginfod-client.c:94:
/nix/store/arrxja9h6jxh9qs5iw3vxcznlkq0w00i-curl-7.87.0-dev/include/curl/curl.h:2836:3: note: declared here
 2836 |   CURLINFO_SIZE_DOWNLOAD
      |   ^~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make[2]: *** [Makefile:638: debuginfod-client.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [Makefile:533: all-recursive] Error 1
make: *** [Makefile:449: all] Error 2
```

### Additional context
Add any other context about the problem here.

### Notify maintainers
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->
@eelco @r-burns @lovek323

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
# warnings removed
$ nix-shell -I nixpkgs=../nixpkgs/ -p nix-info --run "nix-info -m"                                                                                                                                                                                                                              
 - system: `0`
 - host os: `Linux 6.1.1, NixOS, 23.05 (Stoat), 23.05.20221231.9269a0e`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.12.0`
 - channels(root): `""`
 - nixpkgs: `0`
```

--- a/pkgs/development/tools/misc/elfutils/default.nix
+++ b/pkgs/development/tools/misc/elfutils/default.nix
@@ -37,6 +37,18 @@ stdenv.mkDerivation rec {
       url = "https://git.alpinelinux.org/aports/plain/main/elfutils/musl-strndupa.patch?id=2e3d4976eeffb4704cf83e2cc3306293b7c7b2e9";
       sha256 = "sha256-7daehJj1t0wPtQzTv+/Rpuqqs5Ng/EYnZzrcf2o/Lb0=";
     })
+    (fetchpatch {
+      name = "use-curlopt_protocols_str-for-new-libcurl.patch";
+      url = "https://sourceware.org/git/?p=elfutils.git;a=patch;h=6560fb26a62ef135a804357ef4f15a47de3e49b3;hp=a5b07cdf9c491fb7a4a16598c482c68b718f59b9";
+      excludes = [ "debuginfod/ChangeLog" ]; # Doesn't apply
+      sha256 = "sha256-yjeliqojRGvfwbXynmxFGyKqAY7AEr0mbSGQEliYhZ4=";
+    })
+    (fetchpatch {
+      name = "fix-usage-of-deprecated-curlinfo.patch";
+      url = "https://sourceware.org/git/?p=elfutils.git;a=patch;h=d2bf497b12fbd49b4996ccf0744303ffd67735b1;hp=6ecd16410ce1fe5cb0ac5b7c3342c5cc330e3a04";
+      excludes = [ "debuginfod/ChangeLog" ]; # Doesn't apply
+      sha256 = "sha256-zMx/TazM7vXJre2XagIWvwRS8cd8pbzMTmAbpbqZmx0=";
+    })
   ] ++ lib.optionals stdenv.hostPlatform.isMusl [ ./musl-error_h.patch ];
 
   postPatch = ''

